crates/brainfuck_prover/src/brainfuck_air/mod.rs

```diff
@@ -27,7 +27,7 @@ pub struct BrainfuckProof<H: MerkleHasher> {
 /// It includes the common claim values such as the initial and final states
 /// and the claim of each component.
 pub struct BrainfuckClaim {
-    pub memory: memory::component::Claim,
+    pub memory: Claim,
```

Comment:
Then type of the Claim is not enforced by the type here, right?
(Unless we check that `memory.trace == Trace::Memory` it could be a Claim with another Trace type!?)

Reply:
Yes right now this is true

Reply:
just proposed a better solution here